### PR TITLE
add real testimonial on feedbackPage

### PR DESCRIPTION
**Pull request by BaptisteLegat**

### CODE REVIEW

File: src/services/BaseApi.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import * as client from '@FullStackMap/from-a2b';
 import axios from 'axios';
 import Cookies from 'js-cookie';
 
-const basePath: string = 'http://localhost:32769';
+const basePath: string = 'http://api-dev.from-a2b.fr';
```
Comment (Dercraker) [inline src/services/BaseApi.ts]: Use env var

File: src/pages/feedback/FeedbackPage.tsx [hunk 2]
```diff
@@ -12,50 +12,53 @@ import {
...
-  reviews.reduce((acc, review) => acc + review.rating, 0) / reviews.length;
+import { AnoTestimonialsController } from '../../services/BaseApi';
+import { useMutation } from '@tanstack/react-query';
 
 const FeedbackPage = () => {
   const autoplay = useRef(Autoplay({ delay: 5000 }));
 
+  const testimonialsMutation = useMutation({
+    mutationFn: async () => AnoTestimonialsController.getAllTestimonialGET(),
+  });
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Don't use mutation for get [React Query v5](https://tanstack.com/query/latest/docs/framework/react/quick-start)

File: src/pages/feedback/FeedbackPage.tsx [hunk 3]
```diff
@@ -12,50 +12,53 @@ import {
...
+    const fetchData = async () => {
+      try {
+        testimonialsMutation.mutate();
+      } catch (error) {
+        console.error(
+          "Une erreur s'est produite lors de la récupération des données :",
+          error
+        );
+      }
+    };
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: No Try catch here... Use error handler and Notify

File: src/pages/feedback/FeedbackPage.tsx [hunk 4]
```diff
@@ -12,50 +12,53 @@ import {
...
+    fetchData();
+  }, []);
+
+  if (testimonialsMutation.error) {
+    return (
+      <div>
+        Une erreur s'est produite : {testimonialsMutation.error.message}
+      </div>
+    );
+  }
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Replace juste carouselle and no page if Get Testimonials have some errors

File: src/pages/feedback/FeedbackPage.tsx [hunk 5]
```diff
@@ -12,50 +12,53 @@ import {
...
+
+  const reviews = testimonialsMutation.data;
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Export this from query

File: src/pages/feedback/FeedbackPage.tsx [hunk 6]
```diff
@@ -74,24 +77,24 @@ const FeedbackPage = () => {
...
               <Card.Section mah="70%">
                 <Text size="md" m={20}>
-                  {review.comment}
+                  {review.feedBack}
                 </Text>
               </Card.Section>
               <Group mt="auto" justify="space-between">
                 <Group>
-                  <Badge bg="dark">{review.pseudo}</Badge>
+                  <Badge bg="dark">{review.userId}</Badge>
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]:
```suggestion
                  <Badge bg="dark">{review.user.userName}</Badge>
```

**Review by Dercraker — CHANGES_REQUESTED** [review]

**Review by Baptiste-Ferrand — APPROVED** [review]

File: src/pages/feedback/FeedbackPage.tsx [hunk 7]
```diff
@@ -8,53 +8,44 @@ import {
...
+import { useQuery } from '@tanstack/react-query';
+import useNotify, { NotifyDto } from '../../hooks/useNotify';
 
 const FeedbackPage = () => {
   const autoplay = useRef(Autoplay({ delay: 5000 }));
+  const { ErrorNotify } = useNotify();
+
+  const { isPending, isError, data, error } = useQuery({
+    queryKey: ['testimonials'],
+    queryFn: AnoTestimonialsController.getAllTestimonialGET,
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Await this
```suggestion
    queryFn: async () => await AnoTestimonialsController.getAllTestimonialGET,
```

File: src/pages/feedback/FeedbackPage.tsx [hunk 8]
```diff
@@ -8,53 +8,44 @@ import {
...
+  });
+
+  if (error) {
+    ErrorNotify({
+      title: 'Erreur',
+      message: 'Impossible de récupérer les avis',
+    } as NotifyDto);
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Use Catch after request

File: src/pages/feedback/FeedbackPage.tsx [hunk 9]
```diff
@@ -64,42 +61,57 @@ const FeedbackPage = () => {
...
-              </Group>
-            </Card>
-          </Carousel.Slide>
-        ))}
-      </Carousel>
+              </Card>
+            </Carousel.Slide>
+          ))}
+        </Carousel>
+      ) : null}
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Replace null by loader

File: src/pages/feedback/FeedbackPage.tsx [hunk 10]
```diff
@@ -64,42 +61,57 @@ const FeedbackPage = () => {
...
-                <Group>
-                  <Badge bg="dark">{review.pseudo}</Badge>
-                  <Text size="sm" c="dimmed">
-                    {new Date(review.date).toLocaleDateString()}
+      {isPending && (
+        <Center>
+          <Loader size="xl" mt="xl" />
+        </Center>
+      )}
+      {isError && <Text ta="center">Une erreur est survenue</Text>}
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: No this replace by Error Notify

File: src/pages/feedback/FeedbackPage.tsx
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx:1]: Line 127 <FormFeedback /> Disable if user is not connected

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/pages/feedback/FeedbackPage.tsx [hunk 11]
```diff
@@ -1,60 +1,63 @@
...
+      const response = await AnoTestimonialsController.getAllTestimonialGET();
+      return response.data;
+    } catch (error) {
+      handleError();
+    }
+  };
+
+  const { data: reviews } = useQuery({
+    queryKey: ['Testimonials'],
+    queryFn: fetchTestimonials,
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: Use Async func in this queryFn like this : 
```suggestion
    queryFn: async () => await machinController.TrucEndPoint(),
```

File: src/components/feedback/FormFeedback.tsx [hunk 12]
```diff
@@ -25,35 +32,50 @@ export const FormFeedback = () => {
...
+      rate: 0,
+      testimonialDate: new Date().toISOString().split('T')[0],
     },
     validate: zodResolver(feedbackSchema),
   });
 
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  const sendTestimonialMutation = useMutation({
+    mutationFn: async ([id, dto]: [string, AddTestimonialDto]) =>
+      await AnoTestimonialsController.createTestimonialAsyncPOST(id, dto),
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: How to save data with Authentication and Authorization with **Ano**Controller

If you save data with userLogged you must use no **Ano**Controller or you will get 401 http error

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/feedback/FormFeedback.tsx [hunk 13]
```diff
@@ -5,18 +5,25 @@ import { useForm } from '@mantine/form';
...
   const feedbackSchema = z.object({
-    comment: z
+    feedBack: z
       .string()
       .min(10, 'Le commentaire doit contenir au moins 10 caractères'),
-    rating: z
+    rate: z
       .number()
       .min(1, 'La note minimale est de 1')
       .max(5, 'La note maximale est de 5'),
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: Check if the string is not Withe space

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/feedback/FormFeedback.tsx [hunk 14]
```diff
@@ -25,35 +47,50 @@ export const FormFeedback = () => {
...
+  const sendTestimonialMutation = useMutation({
+    mutationFn: async ([id, dto]: [string, AddTestimonialDto]) =>
+      await TestimonialsController.createTestimonialAsyncPOST(id, dto),
+    onSuccess: () => {
+      SuccessNotify({
+        title: 'Avis envoyé',
+        message: 'Votre avis a bien été envoyé, merci !',
+        autoClose: 5000,
+      } as NotifyDto);
+      feedbackForm.reset();
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx:66]: to this in then of async func

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/feedback/FormFeedback.tsx [hunk 15]
```diff
@@ -25,35 +47,50 @@ export const FormFeedback = () => {
...
+    },
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: Invalid Query Cach

**Review by Dercraker — CHANGES_REQUESTED** [review]

**Review by Dercraker — APPROVED** [review]

**Pull request merged**